sw/device/lib/testing/keymgr_dpe_testutils.c

```diff
@@ -0,0 +1,112 @@
+// Copyright lowRISC contributors (OpenTitan project).
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/testing/keymgr_dpe_testutils.h"
+
+#include "sw/device/lib/dif/dif_keymgr_dpe.h"
+#include "sw/device/lib/dif/dif_otp_ctrl.h"
+#include "sw/device/lib/dif/dif_rstmgr.h"
+#include "sw/device/lib/runtime/ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/otp_ctrl_testutils.h"
+#include "sw/device/lib/testing/rstmgr_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/silicon_creator/lib/base/chip.h"
+
+#include "hw/top_darjeeling/sw/autogen/top_darjeeling.h"
+
+status_t keymgr_dpe_testutils_startup(dif_keymgr_dpe_t *keymgr_dpe,
+                                      uint32_t slot_dst_sel) {
+  dif_rstmgr_t rstmgr;
+  dif_rstmgr_reset_info_bitfield_t info;
+
+  TRY(dif_rstmgr_init(
+      mmio_region_from_addr(TOP_DARJEELING_RSTMGR_AON_BASE_ADDR), &rstmgr));
+  info = rstmgr_testutils_reason_get();
+
+  // POR reset.
+  if (info == kDifRstmgrResetInfoPor) {
+    LOG_INFO("Powered up for the first time, lock SECRET2 partition");
```

Comment:
It looks like the SECRET3 locking is missing.

Reply:
Do you know why that would be needed? I can see that the vanilla keymgr also only locks `SECRET2`

Reply:
Ah you're right. I hadn't updated my `integrated_dev` checkout. I've redone the testutils port and this was the only diff, good catch.